cumulus_library/template_sql/utils.py

```diff
@@ -126,3 +151,52 @@ def is_codeable_concept_array_populated(
     if cursor.fetchone() is None:
         return False
     return True
+
+
+def is_code_populated(
+    schema: str,
+    table: str,
+    base_col: str,
+    cursor,
+    allow_partial: bool = True,
+) -> bool:
+    """Check db to see if a bare code exists and is populated.
+
+    Will execute several exploratory queries to see if the column in question
+    can be queried naively.
+
+    :param schema: The schema/database name
+    :param table: The table to query against
+    :param base_col: the place to start validation from.
+        This can be a nested element, like column.object.code
+    :param cursor: a PEP-249 compliant database cursor
+    :allow_partial: If true, codings which do not have fields expected by the library
+        will still be included, and will need to be manually coerced.
+    :returns: a boolean indicating if valid data is present.
+    """
+
+    # if the source column is missing for some reason (i.e. we're dealing with
+    # conversion to FHIR rather than a true FHIR source and it's incomplete),
+    # we'll return false
+    try:
+        query = get_is_table_not_empty_query(table, base_col)
+        cursor.execute(query)
+        if cursor.fetchone() is None:
+            return False
+
+        query = get_column_datatype_query(schema, table, base_col)
+        cursor.execute(query)
+        schema_str = str(cursor.fetchone()[0])
+    except:
+        return False
+    if any(x not in schema_str for x in ["code", "system", "display"]):
+        return False
```

Comment:
nit: again, duplicated code with the above two methods. Feels like a utility method that takes in a list of required fields to check would be able to be shared?